oremda/pipeline.py

```diff
@@ -140,6 +143,25 @@ def set_graph(self, nodes, edges):
             self_node_to_edges.setdefault(output_node.id, set()).add(edge.id)
             self_node_to_edges.setdefault(input_node.id, set()).add(edge.id)
 
+        # Verify that all the required operator input ports have a connection
+        for node in nodes:
+            required_input_ports = set()
+            for port in node.inputs.values():
+                if port.required:
+                    required_input_ports.add(port.name)
```

Comment:
Not a necessary change: this could be a one-liner:
```python
required_input_ports = {port.name for port in node.inputs.values() if port.required}
```